packages/components/pie-modal/src/index.ts

```diff
@@ -154,33 +154,60 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         }
     }
 
-    render () {
-        const {
-            heading,
-            headingLevel = 'h2',
-            size,
-            isFullWidthBelowMid,
-        } = this;
+    public render = () => html`
+        <dialog
+            id="dialog"
+            class="c-modal"
+            size="${this.size}"
+            ?isFullWidthBelowMid=${this.isFullWidthBelowMid}
+            data-test-id="pie-modal">
+            <header>
+                ${this._renderModalHeader()}
+            </header>
+            <article class="c-modal-content c-modal-content--scrollable">
+                <slot></slot>
+            </article>
+            <footer class="c-modal-footer">
+                ${this._renderModalFooter()}
+            </footer>
+        </dialog>`;
 
-        const headingTag = unsafeStatic(headingLevel);
+    /**
+     * Returns the HTML content for the header.
+     * Extracted to keep the render method succinct.
+     * @returns {TemplateResult} HTML template for the modal's header
+     */
+    private _renderModalHeader () : TemplateResult {
+        const headingTag = unsafeStatic(this.headingLevel);
 
         return html`
-            <dialog
-                id="dialog"
-                class="c-modal"
-                size="${size}"
-                ?isFullWidthBelowMid=${isFullWidthBelowMid}>
-                <header>
-                    <${headingTag} class="c-modal-heading">${heading}</${headingTag}>
-                     ${this.isDismissible ? this.renderCloseButton() : nothing}
-                </header>
-                <article class="c-modal-content">
-                    <slot></slot>
-                </article>
-            </dialog>
-        `;
+            <${headingTag} class="c-modal-heading">
+                ${this.heading}
+            </${headingTag}>
+            ${this.isDismissible ? this.renderCloseButton() : nothing}`;
     }
 
+    /**
+     * Returns the HTML content for the footer.
+     * Extracted to keep the render method succinct.
+     * @returns {TemplateResult} HTML template for the modal's footer
+    */
+    private _renderModalFooter = () : TemplateResult => html`
+        <pie-button
```

Comment:
Not entirely necessary but you could store the individual aspects of the button in an array then pass it into one element (instead of having the two bits of markup) as a loop / map? (not tested this tho)

```javascript
private renderModalFooter(): TemplateResult {
  const buttonsValues = [
    { variant: 'primary', type: ... },
    { variant: 'ghost', type: ... }
  ];

  return html`
    ${buttons.map(button => html`
      <pie-button
        variant=${button.variant}....etc}
      >
        ${button.label}
      </pie-button>
    `)}
  `;
}
```

Reply:
This was just setting up the defaults, and I didn't want to make it too complicated.
With the upcoming tickets for the leading and supporting actions, there are some opportunities to try out some stuff like this 👍🏻